waiter/integration/waiter/websocket_integration_test.clj

```diff
@@ -192,6 +198,152 @@
         (finally
           (delete-token-and-assert waiter-url token))))))
 
+(deftest ^:parallel ^:integration-fast test-request-maintenance-mode
+  (testing-using-waiter-url
+    (let [ws-response-atom (atom [])
+          token (str "token-" (rand-name))
+          token-description (assoc (kitchen-request-headers :prefix "")
+                              :authentication "disabled"
+                              :maintenance {"message" "custom maintenance message"}
+                              :metric-group "waiter_ws_test"
+                              :name (rand-name)
+                              :permitted-user "*"
+                              :run-as-user (retrieve-username)
+                              :token token)
+          waiter-headers {"x-waiter-token" token}]
+      (testing "should not upgrade if token is in maintenance mode"
+        (try
+          (let [token-response (post-token waiter-url token-description)]
+            (assert-response-status token-response http-200-ok)
+            (try
+              (let [response-promise (promise)
+                    connection (ws-client/connect!
+                                 (websocket-client-factory)
+                                 (ws-url waiter-url "/websocket-auth")
+                                 (fn [{:keys [in out]}]
+                                   (async/go
+                                     (async/>! out "request-info")
+                                     (swap! ws-response-atom conj (async/<! in))
+                                     (swap! ws-response-atom conj (async/<! in))
+                                     (deliver response-promise :done)
+                                     (async/close! out)))
+                                 {:middleware (fn [_ ^UpgradeRequest request]
+                                                (websocket/add-headers-to-upgrade-request! request waiter-headers))})
+                    [close-code error] (connection->ctrl-data connection)]
+                (is (= :qbits.jet.websocket/error close-code))
+                (is (instance? UpgradeException error))
+                (is (str/includes? (.getMessage error) "Unexpected HTTP Response Status Code: 503 Service Unavailable")))
```

Comment:
Does `custom maintenance message` show up in the response here?

Reply:
For some reason, the message that is sent using the ```ServletUpgradeResponse response``` object is not showing up in the error object here. That maintenance message gets lost somewhere in how the client handles the response